### PR TITLE
Creating an experimental moveit docker image - warehouse / mongo

### PR DESCRIPTION
**Pull request by ugocupcic**
As discussed in https://github.com/ros-planning/moveit_docker/pull/3

### CODE REVIEW

File: .docker/experimental/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# moveit/moveit:kinetic-experimental
+# Based on a moveit source install, adds the mongo driver and the warehouse packages
+
+FROM moveit/moveit:kinetic-source
```
Comment (davetcoleman) [inline .docker/experimental/Dockerfile]: if you build on top of kinetic-source you should use CATKIN_WS = /root/ws_moveit like the underlying container did

File: .docker/experimental/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    cd mongo-cxx-driver && \
+    scons --use-system-boost --prefix=/usr/local/ --full --disable-warnings-as-errors && \
+    ls /usr/local/lib && \
+    # scons install && \
+    rm -rf /var/lib/apt/lists/*
+
+# Download warehouse source
+RUN wstool init . && \
+    wstool set -yu warehouse_ros_mongo --git https://github.com/TheBrewCrew/warehouse_ros_mongo.git -v jade-devel && \
+    wstool set -yu warehouse_ros --git https://github.com/ros-planning/warehouse_ros.git -v jade-devel
```
Comment (davetcoleman) [inline .docker/experimental/Dockerfile]: did you mean to use the jade branch for a kinetic docker?

Comment (ugocupcic) [inline .docker/experimental/Dockerfile]: well there's no kinetic branch for those :S

File: .docker/experimental/Dockerfile (hunk 1 above)
Comment (ugocupcic) [inline .docker/experimental/Dockerfile]: I thought it'd be cleaner to have an overlay? Happy to moveit to the same workspace

Comment (davetcoleman) [inline .docker/experimental/Dockerfile]: ah i see, ok

**Comment by davetcoleman** [comment]
We don't have an easy way to test this in the cloud, but I'm testing the container build locally using:

> docker build -t moveit/moveit:kinetic-experimental .

**Comment by ugocupcic** [comment]
Nod, If you want the prebuilt one I've pushed it to shadowrobot/moveit:kinetic-experimental. I'm rebuilding to test with one workspace only. Will push as soon as it finishes buiding.

**Comment by davetcoleman** [comment]
The image built fine locally!

Should this be cherry-picked to I/J? Seems like indigo had no issue with mongo so it wouldn't be needed there

**Comment by ugocupcic** [comment]
probably jade at least right? For indigo I was under the impression that you don't need all the work to get it working - apt-get install ros...warehouse is enough no?

**Comment by ugocupcic** [comment]
@davetcoleman I think it's ready now? it's building happily with one workspace

**Comment by ugocupcic** [comment]
By the way, you might want to send an email around since the existing moveit configuration would need to be edited to work with the warehouse? Not sure how best to share that info.

**Comment by davetcoleman** [comment]
Thanks @ugocupcic, I've cherry-picked to Jade and turned on the build on [dockerhub](https://hub.docker.com/r/moveit/moveit/~/settings/automated-builds/)

> By the way, you might want to send an email around since the existing moveit configuration would need to be edited to work with the warehouse?

Did the other PR concerning this already get merged in? If so, I think you should make the announcement yourself

**Pull request merged**